test/libsolidity/syntaxTests/types/empty_tuple_lvalue_array.sol

```diff
@@ -0,0 +1,13 @@
+pragma solidity ^0.4.3;
+contract C {
+    function f() private pure {}
+    function a() public {
+        uint x;
+        uint y;
+        (x, y) = [f(), f()];
+    }
+}
+// ----
+// Warning: (146-149): Tuple component cannot be empty.
+// Warning: (151-154): Tuple component cannot be empty.
+// TypeError: (145-155): Type tuple()[2] memory is not implicitly convertible to expected type tuple(uint256,uint256).
```

Comment:
Because of that, I think we can turn into an error for arrays right away.

Reply:
Do you mean assigning it to a tuple should throw an error even before the components are being checked?

Reply:
No, the tuple ast node is used both for tuple expressions and for inline arrays. Empty component should be a warning/error for tuples but an error for arrays.

Reply:
Ok, I see. Will change it accordingly.